src/HttpHeaderTools.cc

```diff
@@ -298,12 +298,7 @@ httpHdrMangle(HttpHeaderEntry * e, HttpRequest * request, HeaderManglers *hms, c
     }
 
     ACLFilledChecklist checklist(hm->access_list, request, nullptr);
-
-    checklist.al = al;
-    if (al && al->reply) {
-        checklist.reply = al->reply.getRaw();
-        HTTPMSGLOCK(checklist.reply);
-    }
+    checklist.updateAle(al);
```

Comment:
This PR uses new updateAle() in this and similar dangerous `HTTPMSGLOCK(checklist.reply)` places. After these changes, more than a dozen places in Squid code still set ALE::al directly instead of calling updateAle(). They all should eventually be upgraded to updateAle() or similar. To avoid mishaps, ALE::al member itself should made private, affecting dozens of readers. All those large/noisy changes are outside this bug-fixing PR scope.